### DISCUSSION TITLE
How to split dotfiles into two remote repos?

### DISCUSSION THREAD

**Discussion by CyrusYip** [twpayne/chezmoi#2780] [Q&A] [answered]
I share most of my dotfiles on a [public repo](https://github.com/CyrusYip/dotfiles), but I want to put some of my private dotfiles to a private GitHub repo.

Say I have `public.conf` and `private.conf` on my local machine. I want `public.conf` in the remote public repo and `private.conf` in the remote private repo.

---

**Solution to manage public files and private files** (based on the [answer](https://github.com/twpayne/chezmoi/discussions/2780#discussioncomment-5011557) of @halostatue ): Use `--source` option for private dotfiles.

1. Create a public repo and follow the official docs (https://www.chezmoi.io/user-guide/setup/)
1. Create a private repo
2. Run `chezmoi init --source ~/.local/share/chezmoi-private --ssh https://github.com/CyrusYip/repo-name`
3. Manage private files with `chezmoi --source ~/.local/share/chezmoi-private <command>`
4. Manage public files with `chezmoi <command>`.
5. Using alias makes you type less (`alias cho='chezmoi --source ~/.local/share/chezmoi-private'`).

**Comment by halostatue** [comment, marked as answer]
This is mostly answered with the FAQ "[Can Chezmoi Support Multiple Sources or Multiple Source States?](https://www.chezmoi.io/user-guide/frequently-asked-questions/design/#can-chezmoi-support-multiple-sources-or-multiple-source-states)".

It’s worth noting that, for the most part, all you need is the `--config` option as all other file and path options (*except* `--persistent-state`) can be configured in your `chezmoi.$FORMAT` configuration file, simplifying the commands, functions, or aliases required to execute this.

There are a couple of other options that you can explore:

- You can put your private configuration in an external and then have an `after_` script symlink files to the appropriate location. The downsides here are that the private configuration wouldn’t be templated or automatically placed.

- You can keep your configuration files public, but iterate over both public and private configuration sources (mine is 1Password), like I do for [`~/.aws/credentials`](https://github.com/halostatue/dotfiles/blob/main/private_dot_aws/private_credentials.tmpl) with the relevant configuration in [`chezmoi.$FORMAT`](https://github.com/halostatue/dotfiles/blob/main/.chezmoi.toml.tmpl#L9-L24).

**Reply by CyrusYip** [reply]
Thanks, I will try and see which method works for me.

**Reply by CyrusYip** [reply]
Using `--source` option is enough for me. I wrote the solution on first post https://github.com/twpayne/chezmoi/discussions/2780#discussion-4868605 .

**Comment by Lockszmith-GH** [comment]
I've written about this [here](https://github.com/twpayne/chezmoi/discussions/2574) and [here](https://github.com/twpayne/chezmoi/issues/2771#issuecomment-1431726265)
I will be happy to share more details if this is of any interest.
